plugins/kubernetes/app/models/kubernetes/cluster.rb

```diff
@@ -26,7 +26,7 @@ class Cluster < ActiveRecord::Base
     before_destroy :ensure_unused
 
     def client(type)
-      (@client ||= {})[type] ||= begin
+      (@client ||= {})[[Thread.current.object_id, type]] ||= begin
```

Comment:
alternatively a mutex/lock might have worked...

Reply:
- the client most likely is not threadsafe internally, so 1 per thread is safer
 - the race condition is not here but when resolving the "is get_deployments defined" so I'd have to hack a mutex around that